src/main/java/pixelperfect/Spaceship.java

```diff
@@ -15,6 +16,7 @@
 public class Spaceship {
 
 	private static final Spaceship ship = new Spaceship();
+	private RouteGenerator routeGenerator;
```

Comment:
I don't think this is a good idea: the Route itself must know how it should be created / generated.